src/messages/manifestjson.js

```diff
@@ -35,6 +35,16 @@ export const MANIFEST_FIELD_INVALID = {
   file: MANIFEST_JSON,
 };
 
+export const MANIFEST_BAD_PERMISSION = {
+  code: 'MANIFEST_BAD_PERMISSION',
+  legacyCode: null,
+  message: _('The permission type is unsupported.'),
+  description: _(singleLineString`Permissions must be strings; object
```

Comment:
I believe it's for apps, not extensions. See https://github.com/mozilla/addons-server/issues/2726#issuecomment-228899614